crates/aptos/src/node/mod.rs

```diff
@@ -693,3 +695,61 @@ impl CliCommand<()> for RunLocalTestnet {
         Ok(())
     }
 }
+
+/// Update the current validator's network and fullnode addresses
+#[derive(Parser)]
+pub struct UpdateValidatorNetworkAddresses {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+
+    #[clap(flatten)]
+    pub(crate) operator_args: OperatorArgs,
+
+    #[clap(flatten)]
+    pub(crate) validator_config_args: ValidatorConfigArgs,
+}
+
+#[async_trait]
+impl CliCommand<Transaction> for UpdateValidatorNetworkAddresses {
+    fn command_name(&self) -> &'static str {
+        "UpdateValidatorNetworkAddresses"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<Transaction> {
+        let address = self
+            .operator_args
+            .address(&self.txn_options.profile_options)?;
+
+        let validator_config = self.validator_config_args.read_validator_config()?;
+        let (
+            validator_network_public_key,
+            full_node_network_public_key,
+            validator_host,
+            full_node_host,
+        ) = self
+            .validator_config_args
+            .get_network_configs(&validator_config)?;
+        let validator_network_addresses =
+            vec![validator_host.as_network_address(validator_network_public_key)?];
+        let full_node_network_addresses =
+            match (full_node_host.as_ref(), full_node_network_public_key) {
+                (Some(host), Some(public_key)) => vec![host.as_network_address(public_key)?],
+                _ => vec![],
+            };
+
+        self.txn_options
+            .submit_script_function(
+                AccountAddress::ONE,
+                "stake",
+                "update_network_and_fullnode_addresses",
+                vec![],
+                vec![
+                    bcs::to_bytes(&address)?,
+                    // Double BCS encode, so that we can hide the original type
+                    bcs::to_bytes(&bcs::to_bytes(&validator_network_addresses)?)?,
+                    bcs::to_bytes(&bcs::to_bytes(&full_node_network_addresses)?)?,
```

Comment:
@gregnazario We had a fun dive last time regarding this. Is this a single address or a vector again?

Reply:
This is a double BCS encoded vector of addresses